imports/server/announcements.ts

```diff
@@ -4,6 +4,40 @@ import { Roles } from 'meteor/nicolaslopezj:roles';
 import Ansible from '../ansible';
 import Announcements from '../lib/models/announcements';
 import PendingAnnouncements from '../lib/models/pending_announcements';
+import Profiles from '../lib/models/profiles';
+
+Meteor.publish('announcements.pending', function () {
+  if (!this.userId) {
+    throw new Meteor.Error(401, 'Unauthenticated');
+  }
+
+  const pas = PendingAnnouncements.find({ user: this.userId });
+  const announcementIds = [...new Set(pas.map(pa => pa.announcement))];
+  const announcements = Announcements.find({ _id: { $in: announcementIds } });
+
+  const creatorIds = [...new Set(announcements.map(a => a.createdBy))];
+  const creators = Profiles.find({ _id: { $in: creatorIds } }, { fields: { displayName: 1 } });
+
+  return [pas, announcements, creators];
```

Comment:
This has the same issue as `announcements.all` below -- the `PendingAnnoucements` collection available client-side will get updated reactively later, but `Annoucements` and `Profiles` will not.